py-polars/polars/interchange/utils.py

```diff
@@ -0,0 +1,55 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import polars as pl
+from polars.interchange.protocol import DtypeKind, Endianness
+
+if TYPE_CHECKING:
+    from polars.interchange.protocol import Dtype
+    from polars.type_aliases import PolarsDataType
+
+
+def polars_dtype_to_dtype(dtype: PolarsDataType) -> Dtype:
```

Comment:
Maybe this is premature optimization. But if we call this often we could make a lazy instantiated dict from this. As we have a long if else call chain before we hit.

Reply:
Yeah, this code is not pretty. But dictionaries don't really work well here because some datatypes are classes and others are instantiated.

This is also the reason I created this issue: https://github.com/pola-rs/polars/issues/6163

I don't really see a pretty fix until we have overhauled our type system.

Reply:
Updated the code to be more performant. It now uses a dictionary and then handles the instantiated types (Datetime, Duration) separately.